pkg/daemon/config/config.go

```diff
@@ -0,0 +1,48 @@
+//go:build windows
+
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"io/ioutil"
+
+	"k8s.io/client-go/rest"
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/klog/v2"
+)
+
+// FromServiceAccount uses service account credentials to create a client config able to authenticate with a cluster
+func FromServiceAccount(apiServerURL, caFile, tokenFile string) (*rest.Config, error) {
+	token, err := ioutil.ReadFile(tokenFile)
+	if err != nil {
+		klog.Errorf("error reading token file: %v", err)
```

Comment:
Knowing that this was a part of the previous version of this function
Is the logging required here if we can return the same error?